File: Wms.Api/Dto/Category/Category.cs
 using Wms.Api.Entities; 

namespace Wms.Api.Dto.Category
{
    public class CategoryDetailsDto : CreatedChangedEntity
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
    }

    public class CategoryCreateUpdateDto
    {
        public required string Name { get; set; }
    }

    public class CategorySearchDto : PagedRequestAbstractDto
    {
        public required string Search { get; set; }
    }
}

File: Frontend/src/interfaces/v12/category/category.ts
import { CreatedChangedEntity } from "interfaces/v12/CreatedChangedEntity";
import { guid } from "types/guid";
import { PagedRequestAbstractDto } from "../pagedRequest/pagedRequestAbstractDto";

export interface CategoryDetailsDto extends CreatedChangedEntity {
  id: guid;
  name: string;
}

export interface CategoryCreateUpdateDto {
  name: string;
}

export interface CategorySearchDto extends PagedRequestAbstractDto {
  search: string;
}
